chips/nrf52/src/ieee802154_radio.rs

```diff
@@ -767,128 +797,250 @@ impl<'a> Radio<'a> {
         buffer
     }
 
-    // TODO: Theres an additional step for 802154 rx/tx handling
+    fn crc_check(&self) -> Result<(), ErrorCode> {
+        if self.registers.crcstatus.is_set(Event::READY) {
+            Ok(())
+        } else {
+            Err(ErrorCode::FAIL)
+        }
+    }
+
+    // TODO: RECEIVING ACK FOR A SENT TX IS NOT IMPLEMENTED //
+    // As a general note for the interrupt handler, event registers must still be cleared even when
+    // hardware shortcuts are enabled.
     #[inline(never)]
     pub fn handle_interrupt(&self) {
         self.disable_all_interrupts();
 
-        if self.registers.event_ready.is_set(Event::READY) {
-            self.registers.event_ready.write(Event::READY::CLEAR);
-            self.registers.event_end.write(Event::READY::CLEAR);
-            if self.transmitting.get()
-                && self.registers.state.get() == nrf5x::constants::RADIO_STATE_RXIDLE
-            {
-                self.registers.task_ccastart.write(Task::ENABLE::SET);
-            } else {
-                self.registers.task_start.write(Task::ENABLE::SET);
-            }
-        }
+        let mut start_task = false;
+        let mut rx_init = false;
 
-        if self.registers.event_framestart.is_set(Event::READY) {
-            self.registers.event_framestart.write(Event::READY::CLEAR);
-        }
-
-        //   IF we receive the go ahead (channel is clear)
-        // THEN start the transmit part of the radio
-        if self.registers.event_ccaidle.is_set(Event::READY) {
-            self.registers.event_ccaidle.write(Event::READY::CLEAR);
-            self.registers.task_txen.write(Task::ENABLE::SET)
-        }
+        match self.state.get() {
+            // It should not be possible to receive an interrupt while the tracked radio state is OFF
+            RadioState::OFF => panic!("Received interrupt while off"),
```

Comment:
This is a place where I think just ignoring the interrupt, rather than panicking, makes more sense, since panics _do_ incur some overhead (code size, particularly for the format string).

Reply:
If this condition ever occurs, we have diverged from the state machine and some bug exists in the driver. From a "handling the interrupt perspective" I agree that it makes sense to just ignore here. 

Is there some other way to still prevent a "silent fail" if our state diverges from the state machine?

Reply:
Not that I can think of. The trouble is that `panic` is both catastrophic and has high size overhead. If ignoring completely is unpalatable (particularly for a still early implementation), perhaps replacing this with a `debug!` might be better. This won't fail "silently" but is also might be easier (though not currently supported) to "turn off" in production.

Reply:
For what it is worth, the only place I know of where we handle something like this is in Process within the kernel, where we have a error type which is basically "kernel_bug" that is returned in all of these cases that should never happen but the compiler cannot verify that.

But, these type of state machine errors don't seem to happen in practice. Either there are bugs and even simple testing identifies them, or they are correct and panic is never called.

Reply:
Thank you for the input. I'll go ahead and add a debug! statement instead of the panic.